Sources/PromiseKit/Promise.swift

```diff
@@ -56,12 +56,10 @@ public class Promise<Value>: Future<Value, Error> {
   }
   
   public func resolve(with value: Value, on dispatchQueue: DispatchQueue? = .main) {
-    guard isAwaiting else { return }
```

Comment:
Moving this check into the `Future.setResult(_,dispatchQueue:)` serves two purposes:
 - No need to acquire the lock twice, increasing the performance.
 - Potentially fixes a race condition, where a timely interrupt might suspend the thread right after `isAwaiting` returns, entering the critical section inside `setResult` twice.